apps/sms/js/thread_list_ui.js

```diff
@@ -393,17 +394,24 @@ var ThreadListUI = {
     }
   },
 
-  markReadUnread: function thlui_markReadUnread(selected, isRead) {
-    selected.forEach((id) => {
+  markReadUnreadUI: function thlui_markReadUnreadUI(selected, isRead) {
+    function filterIds(id) {
       var thread  = Threads.get(id);
       var markable = thread && (!thread.hasDrafts || isRead);
 
       if (markable) {
         thread.unreadCount = isRead ? 0 : 1;
-        this.mark(thread.id, isRead ? 'read' : 'unread');
-
-        MessageManager.markThreadRead(thread.id, isRead);
+        ThreadListUI.mark(thread.id, isRead ? 'read' : 'unread');
       }
+      return markable;
+    }
+
+    ThreadListUI.markReadUnread(selected.filter(filterIds), isRead);
+  },
+
+  markReadUnread: function thlui_markReadUnread(threadToMark, isRead) {
+    threadToMark.forEach((id) => {
+        MessageManager.markThreadRead(id, isRead);
     });
 
     this.cancelEdit();
```

Comment:
we should call cancelEdit after the UI change, not after the "real" change (especially considering your next patch).